internal/engine/filter/filter.go

```diff
@@ -67,6 +68,67 @@ func (f Filter) String() string {
 	return w.String()
 }
 
+func indent(query string, level int) string {
+	return strings.TrimRight(indentInternal(query, level), "\n ")
+}
+
+func indentInternal(query string, level int) string {
+	var out bytes.Buffer
+	firstChar := true
+	sawColon := true
+	inQuotes := false
+
+	for j := 0; j < level; j++ {
+		out.WriteString("  ")
+	}
+	for i, r := range query {
+		switch r {
+		case ' ':
+			if inQuotes {
+				out.WriteByte(' ')
+				continue
+			}
+			if sawColon {
+				continue
+			}
+			if !firstChar {
+				out.WriteByte('\n')
+			}
+
+			for j := 0; j < level; j++ {
+				out.WriteString("  ")
+			}
+		case '{', '(':
+			out.WriteRune(r)
+			out.WriteByte('\n')
+			return out.String() + indent(query[i+1:], level+1)
+		case '}', ')':
+			if !firstChar {
+				out.WriteByte('\n')
+				for j := 0; j < level-1; j++ {
+					out.WriteString("  ")
+				}
+			}
+			out.WriteByte(query[i])
+			out.WriteByte('\n')
+			return out.String() + indent(query[i+1:], level-1)
+		case ':':
+			out.WriteByte(':')
+			sawColon = true
+		case '"':
+			inQuotes = !inQuotes
+			out.WriteByte('"')
+
+		default:
+			sawColon = false
```

Comment:
Should this be outside of default? I'm thinking that also `{}()" ` are not colons.